### DISCUSSION TITLE
v3 Upgrade guide: Include payload column type change

### DISCUSSION THREAD

**Discussion by andzandz** [spatie/laravel-stripe-webhooks#121] [Q&A] [answered]
Hi, nice package you have here.

We had a nasty surprise recently due to the the webhook_calls payload column changing from type text to type json between v2 and v3.

In v2, the payload column is of type text:
https://github.com/spatie/laravel-webhook-client/blob/v2/database/migrations/create_webhook_calls_table.php.stub

In v3, the payload column has changed to type json:
https://github.com/spatie/laravel-webhook-client/blob/main/database/migrations/create_webhook_calls_table.php.stub

The v3 upgrade guide correctly tells you to add the url and headers columns, but does not mention the payload column change:
https://github.com/spatie/laravel-stripe-webhooks/blob/main/UPGRADING.md#heres-an-example-how-your-migration-should-look

StripeWebhookProfile uses a json operator which will fail (HTTP 500) if the payload column is not changed to type json:
https://github.com/spatie/laravel-stripe-webhooks/blob/main/src/StripeWebhookProfile.php

To fix it, adding this to the add_columns_to_webhook_calls upgrade migration works on MySQL:

```
$table->json('payload')->change();
```

But we use PostgreSQL, and unsurprisingly this doesn't work. This raw SQL works:

```
DB::statement('alter table webhook_calls alter column payload type json using payload::json');
```

**Comment by freekmurze** [comment, marked as answer]
We currently only support MySQL. Feel free to PR this suggestion to the docs for Postgres users.

**Reply by andzandz** [reply]
I have created a PR for this which will be sufficient for MySQL: https://github.com/spatie/laravel-stripe-webhooks/pull/122
